npm/webpack-batteries-included-preprocessor/index.js

```diff
@@ -131,23 +132,51 @@ const getDefaultWebpackOptions = () => {
         loader: require.resolve('coffee-loader'),
       }],
     },
+    plugins: [
+      new webpack.ProvidePlugin({
+        Buffer: ['buffer', 'Buffer'],
+        process: 'process/browser',
+      }),
+    ],
     resolve: {
       extensions: ['.js', '.json', '.jsx', '.mjs', '.coffee'],
-      alias: {
-        'child_process': require.resolve('./empty'),
-        'cluster': require.resolve('./empty'),
-        'console': require.resolve('./empty'),
-        'dgram': require.resolve('./empty'),
-        'dns': require.resolve('./empty'),
-        'fs': require.resolve('./empty'),
-        'http2': require.resolve('./empty'),
-        'inspector': require.resolve('./empty'),
-        'module': require.resolve('./empty'),
-        'net': require.resolve('./empty'),
-        'perf_hooks': require.resolve('./empty'),
-        'readline': require.resolve('./empty'),
-        'repl': require.resolve('./empty'),
-        'tls': require.resolve('./empty'),
+      fallback: {
+        assert: require.resolve('assert/'),
+        buffer: require.resolve('buffer/'),
+        child_process: false,
+        cluster: false,
+        console: false,
+        constants: require.resolve('constants-browserify'),
+        crypto: require.resolve('crypto-browserify'),
+        dgram: false,
+        dns: false,
+        domain: require.resolve('domain-browser'),
+        events: require.resolve('events/'),
+        fs: false,
+        http: require.resolve('stream-http'),
+        https: require.resolve('https-browserify'),
+        http2: false,
+        inspector: false,
+        module: false,
+        net: false,
+        os: require.resolve('os-browserify/browser'),
+        path: require.resolve('path-browserify'),
+        perf_hooks: false,
+        punycode: require.resolve('punycode/'),
+        process: require.resolve('process/browser'),
+        querystring: require.resolve('querystring-es3'),
+        readline: false,
+        repl: false,
+        stream: require.resolve('stream-browserify'),
+        string_decoder: require.resolve('string_decoder/'),
+        sys: require.resolve('util/'),
+        timers: require.resolve('timers-browserify'),
```

Comment:
Oh, maybe we do polyfill them all. My previous comment may be irrelevant.

Big breaking change here - who knows if these implementations match whatever webpack 4 was doing.

Tracking down the right polyfill here must have been complex.

Reply:
> Oh, maybe we do polyfill them all. My previous comment may be irrelevant.

> Big breaking change here - who knows if these implementations match whatever webpack 4 was doing.

This for sure breaks the `webpack-batteries-included-preprocessor` . In this case, we polyfill most of them and are fairly explicit about the ones we don't in the config. We also have a few tests in place since just about everything goes through the preprocessor. The tests that we do check for the polyfills are [this system test in plugins_spec.js](https://github.com/cypress-io/cypress/blob/develop/system-tests/test/plugins_spec.js#L213) which uses [this spec](https://github.com/cypress-io/cypress/blob/develop/system-tests/projects/e2e/cypress/e2e/node_builtins.cy.js) and checks for just about every polyfill in this [gist](https://gist.github.com/ef4/d2cf5672a93cf241fd47c020b9b3066a#list-of-polyfill-packages-that-were-used-in-webpack-4) except for the ones starting with `_`, in particular the `readable-stream` dependencies. There is also [this unit test](https://github.com/cypress-io/cypress/blob/develop/npm/webpack-batteries-included-preprocessor/test/fixtures/node_builtins_spec.js) that pretty much tests the same thing. At least for the built ins we are want to convey we support we should be covered.

> Tracking down the right polyfill here must have been complex.

this [gist](https://gist.github.com/ef4/d2cf5672a93cf241fd47c020b9b3066a#list-of-polyfill-packages-that-were-used-in-webpack-4) for sure helped to figure out what was going on